### PR TITLE
[AzureMonitorDistro] remove vendored instrumentation libs (AspNetCore & Http)

### PR DESCRIPTION
**Pull request by TimothyMothra**
[OpenTelemetry.Instrumentation.AspNetCore](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.AspNetCore) and [OpenTelemetry.Instrumentation.Http](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.Http) have both released stable versions. We can now remove the internal vendored copies of the beta librares and ship using package references instead.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,10 @@
 
 ### Other Changes
 
+- Updated this package to include package references to the now stable versions of `OpenTelemetry.Instrumentation.AspNetCore` and `OpenTelemetry.Instrumentation.Http`.
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:13]: Make it as different points on vendorded code removal and adding package references.

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md]: updated. I tried to match the language we used in previous releases.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,13 @@
 
 ### Other Changes
 
+- Removed the code of internal vendored instrumentation libraries `OpenTelemetry.Instrumentation.AspNetCore` and `OpenTelemetry.Instrumentation.Http`.
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:13]: As we are adding these 2 points, we may need to explain the impacts. If there are no impact, let us call that out.

**Comment by rajkumar-rangaraj** [comment]
Did you test these changes locally? We have a reflection logic in OpenTelemetryBuilderExtensions.cs that also needs to be removed.

**Comment by TimothyMothra** [comment]
> Did you test these changes locally? We have a reflection logic in OpenTelemetryBuilderExtensions.cs that also needs to be removed.

not yet. investigating why the build is failing, it's compiling locally.

File: eng/Packages.Data.props [hunk 3]
```diff
@@ -148,6 +148,8 @@
   <ItemGroup Condition="$(MSBuildProjectName.StartsWith('Azure.Monitor.OpenTelemetry'))">
     <!-- OpenTelemetry dependency approved for Azure.Monitor.OpenTelemetry.Exporter package only -->
     <PackageReference Update="OpenTelemetry" Version="1.6.0" />
+    <PackageReference Update="OpenTelemetry.Instrumentation.AspNetCore" Version="1.6.0" />
```
Comment (vishweshbankwar) [inline eng/Packages.Data.props:151]: Can we not do 1.7.0?

Comment (TimothyMothra) [reply eng/Packages.Data.props:151]: I thought you had asked us to do a release first with 1.6.0 before upgrading to 1.7.0.
If I'm misremembering, we can upgrade to 1.7.0 this release.

**Pull request closed without merging**